src/content/reference/react/use-server.md

```diff
@@ -24,34 +24,180 @@ canary: true
+  <form action={requestUsername}>
+    <input type="text" name="username" />
+    <button type="submit">Request</button>
+  </form>
+}
+```
+
+In this example `requestUsername` is a server action passed to a `<form>`. When a user submits this form, there is a network request to the server function `requestUsername`. When calling a server action in a form, React will supply the form's <CodeStep step={1}>[FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData)</CodeStep> as the first argument to the server action.
+
+By passing a server action to the form `action`, React can [progressively enhance](https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement) the form. This means that forms can be submitted before the JavaScript bundle is loaded.
+
+#### Handling return values in forms {/*handling-return-values*/}
+
+In the username request form, there might be the chance that a username is not available. `requestUsername` should tell us if it fails or not. How can the form receive the return value of the server action?
+
+Here is where you can use [`useFormState`](/reference/react-dom/hooks/useFormState), a Hook designed for receiving a server action return value which supports progressive enhancement.
+
+```js
+// requestUsername.js
+'use server';
+
+export default async function requestUsername(formData) {
+  const username = formData.get('username');
+  if (canRequest(username)) {
+    // ...
+    return 'successful';
+  }
+  return 'failed';
+}
+```
+
+```js {4,8}, [[2, 2, "'use client'"]]
+// UsernameForm.js
+'use client';
+
+import {useFormState} from 'react-dom';
+import requestUsername from './requestUsername';
+
+function UsernameForm() {
+  const [returnValue, action] = useFormState(requestUsername, 'n/a');
+
+  return (
+    <>
+      <form action={action}>
+        <input type="text" name="username" />
+        <button type="submit">Request</button>
+      </form>
+      <p>Last submission request returned: {returnValue}</p>
+    </>
+  );
+}
+```
+
+Note that `useFormState` can only be called in <CodeStep step={1}>[client code](/reference/react/use-client)</CodeStep>.
+
+### Calling a server action outside of `<form>` {/*calling-a-server-action-outside-of-form*/}
+
+Server actions are exposed server endpoints and can be called anywhere in client code.
+
+If you call a server action outside of a [form](/reference/react-dom/components/form), you'll need to call the server action in a [transition](/reference/react/useTransition). Server actions in forms will automatically handle this for you.
+
+```js {8-11}
+import incrementLike from './actions';
+import {useState, useTransition} from 'react';
+
+function LikeButton() {
+  const [pending, startTransition] = useTransition();
+  const [likeCount, setLikeCount] = useState(0);
+
+  const onClick = () => startTransition(async () => {
+    // `incrementLike` is a server action
+    const currentCount = await incrementLike();
+    setLikeCount(currentCount);
+  });
```

Comment:
for the docs I think this is easier to follow as a block

```suggestion
  const onClick = () => {
    startTransition(async () => {
      // `incrementLike` is a server action
      const currentCount = await incrementLike();
      setLikeCount(currentCount);
    })
  };
```

(and makes clear the return value of startTransition isn't needed)